test/test_Quaternion.jl

```diff
@@ -24,6 +24,25 @@ for _ in 1:10, T in (Float32, Float64, Int32, Int64)
     test_multiplicative(c1, c2, +, Quaternion)
 end
 
+let # test promotions and equalities
```

Comment:
It's more useful to declare a testset:
```suggestion
@testset "promotions and equalities" begin
```

Reply:
Also, can you create a testset for testing the conversions directly instead of just through promotions? The current tests here miss the following conversions, and this way we guarantee to never miss a conversion in the tests.
```julia
convert(::Type{DualQuaternion{T}}, d::Dual) where {T}
convert(::Type{Octonion{T}}, z::Complex) where {T}
```